packages/core/src/actions.ts

```diff
@@ -505,3 +517,115 @@ export function escalate<
     }
   );
 }
+
+export function decide<TContext, TEvent extends EventObject>(
+  conds: DecideConditon<TContext, TEvent>[]
+): DecideAction<TContext, TEvent> {
+  return {
+    type: ActionTypes.Decide,
+    conds
+  };
+}
+
+export function resolveActions<TContext, TEvent extends EventObject>(
+  machine: StateNode<TContext, any, TEvent>,
+  currentState: State<TContext, TEvent> | undefined,
+  currentContext: TContext,
+  _event: SCXML.Event<TEvent>,
+  actions: ActionObject<TContext, TEvent>[]
+): [ActionObject<TContext, TEvent>[], TContext] {
+  const [assignActions, otherActions] = partition(
+    actions,
+    (action): action is AssignAction<TContext, TEvent> =>
+      action.type === actionTypes.assign
+  );
+
+  let updatedContext = assignActions.length
+    ? updateContext(currentContext, _event, assignActions, currentState)
+    : currentContext;
+
+  const resolvedActions = flatten(
+    otherActions.map(actionObject => {
+      switch (actionObject.type) {
+        case actionTypes.raise:
+          return resolveRaise(actionObject as RaiseAction<TEvent>);
+        case actionTypes.send:
+          const sendAction = resolveSend(
+            actionObject as SendAction<TContext, TEvent>,
+            updatedContext,
+            _event,
+            machine.options.delays
+          ) as ActionObject<TContext, TEvent>; // TODO: fix ActionTypes.Init
+
+          if (!IS_PRODUCTION) {
+            // warn after resolving as we can create better contextual message here
+            warn(
+              !isString(actionObject.delay) ||
+                typeof sendAction.delay === 'number',
+              // tslint:disable-next-line:max-line-length
+              `No delay reference for delay expression '${actionObject.delay}' was found on machine '${machine.id}'`
+            );
+          }
+
+          return sendAction;
+        case actionTypes.log:
+          return resolveLog(
+            actionObject as LogAction<TContext, TEvent>,
+            updatedContext,
+            _event
+          );
+        case actionTypes.decide: {
+          const decideAction = actionObject as DecideAction<TContext, TEvent>;
+          const matchedActions = decideAction.conds.find(condition => {
+            const guard = toGuard(condition.cond, machine.options.guards);
+            return (
+              !guard ||
+              evaluateGuard(
+                machine,
+                guard,
+                updatedContext,
+                _event,
+                currentState as any
+              )
+            );
+          })?.actions;
```

Comment:
Will this work? I've always had problems getting the new `foo?.bar` syntax to work with our current setup.

Reply:
Well, CI is green so it works. Maybe there was some intermediate state with some older version of prettier (or some other tool) which had no support for this syntax yet. Or maybe it's not working on next? If you encounter any problems with this on any branch then let me know and I will investigate